### ISSUE TITLE
Different colon symbols

### ISSUE THREAD

**Issue by jonaprieto** [anoma/juvix#953]
The following code type-checks.
If one comments the code exactly below part-C, the file does not type-check.
The reason is due to that, in part-D, the second occurrence of : is not the same symbol
as the first occurrence in the same line nor in part-B nor. This makes the rest of any other type annotation
ambiguous.

```
$ juvix version
Juvix version 0.0.0-0db89d4
$ juvix Colon
mod Colon where

-- part-A
let A_ = %Builtin.Type

-- part-B
sig B : A_ 1 
let B = A_ 0

-- part-C
let (:) = %Builtin.Colon
declare infixr (:) 2

-- part-D
sig C : (D : ty) 
let C = B
```

In order to avoid this issue, I suggest to forbidden the usage of the
colon and treat it as a reserved symbol.

**Comment by cwgoes** [comment]
Yeah, let's just take colon as reserved. Any objections @mariari?

**Comment by mariari** [comment]
> Yeah, let's just take colon as reserved. Any objections @mariari?

We can make it reserved for now. Though we should probably get on the new syntax soon, there `:` should just be a function like any other.

Here the issue just comes from ML syntax punning what `:` means more so than anything else, as `sig` requires 1 fixed `:` eraly on.
